File: Day 2/0 Linear Search - assignment/solution.cpp

vector<int> FindNumbersInRange(vector<int>& arr, int low, int high, int targetNum) {
    
    vector<int> result;

    for(int i = low; i<= high ; i++ ){

        if(arr[i] ==targetNum )
        {
            result.push_back(num);
        }        
    }

    // for (int num : arr) {
    //     if (num >= low && num <= high) {
    //         result.push_back(num);
    //     }
    // }
    return result;
}

File: Day 2/0 Linear Search - assignment/solution.cs
int[] FindNumbersInRange(int[] arr, int low, int high, int targetNum) {

    //edge cases
    if(low < 0)
        low = 0;

    //edge cases
    if(high > arr.Length - 1)
        high = arr.Length -1;    
    
    List<int> result = new List<int>();

    for(int i = low; i<= high ; i++ ){

        if(arr[i] ==targetNum )
        {
            result.Add(num);
        }        
    }

    // foreach (int num in arr) {

    //     if (num >= low && num <= high) {
           
    //     }
    // }
    return result.ToArray();
}
